software/nosql/src/main/java/brooklyn/entity/nosql/riak/RiakNodeSshDriver.java

```diff
@@ -156,6 +157,34 @@ public void install() {
                 .add("ln -s `which riak-admin` " + Urls.mergePaths(installBin, "riak-admin"))
                 .build();
     }
+    
+    private List<String> installPackageCloud() {
+        return ifExecutable0Else1("yum", installDebianBased(), installRpmBased());
+    }
+
+    private ImmutableList<String> installDebianBased() {
+        ImmutableList.Builder<String> commands = ImmutableList.<String>builder();
+        commands.add("curl https://packagecloud.io/install/repositories/basho/riak/script.deb | sudo bash");
+        commands.add("sudo apt-get install --assume-yes riak");
+        return commands.build();
+    }
+    
+    private ImmutableList<String> installRpmBased() {
+        ImmutableList.Builder<String> commands = ImmutableList.<String>builder();
+        commands.add("curl https://packagecloud.io/install/repositories/basho/riak/script.rpm | sudo bash");
+        commands.add("sudo yum install -y riak");
+        return commands.build();
+    }
+
+    private static ImmutableList<String> ifExecutable0Else1(String command, List<String> ifTrue, List<String> otherwise) {
```

Comment:
Name is a bit confusing, particularly compared to `BashCommands.ifExecutableElse1` and `BashCommands.ifExecutableElse0` - those are semantically like your impl with the `otherwise` being to just return 0 or 1 respectively.

Perhaps `ifExecutableElse` would be better, as there is no 0 or 1 involved?